events.go

```diff
@@ -215,22 +251,24 @@ var NullEventStream = EventStreamFunc(func(event Event) error {
 	return nil
 })
 
-// AsyncEventStream wraps an EventStream to publish events asynchronously in a
-// goroutine.
+// AsyncEventStream wraps an array of EventStreams to publish events
+// asynchronously in a goroutine
 type AsyncEventStream struct {
 	EventStream
-	events chan Event
+
+	streams []EventStream
+	events  chan Event
 }
 
 // AsyncEvents returns a new AsyncEventStream that will buffer upto 100 events
 // before applying backpressure.
-func AsyncEvents(e EventStream) *AsyncEventStream {
-	s := &AsyncEventStream{
-		EventStream: e,
-		events:      make(chan Event, 100),
+func AsyncEvents(s []EventStream) *AsyncEventStream {
```

Comment:
https://github.com/hashicorp/go-multierror seems to handle multierrors cleanly so i went with that instead of reimplementing something.